packages/manager/apps/pci-kubernetes/src/helpers/node-pool.ts

```diff
@@ -0,0 +1,58 @@
+import { ANTI_AFFINITY_MAX_NODES, NODE_RANGE } from '@/constants';
+import { isMonoDeploymentZone } from '.';
+import { NodePoolState } from '@/types';
+import { TRegionInformations } from '@/types/region';
+
+export const exceedsMaxNodes = (quantity: number) => quantity > NODE_RANGE.MAX;
+
+export const zoneAZisChecked = (
+  regionInformations: TRegionInformations,
+  nodePoolState: NodePoolState,
+) =>
+  isMonoDeploymentZone(regionInformations?.type) ||
+  !!nodePoolState.selectedAvailabilityZone;
+
+export const isScalingValid = (nodePoolState: NodePoolState) => {
+  if (!nodePoolState.scaling) return true;
+
+  const { isAutoscale, quantity } = nodePoolState.scaling;
+  const { desired, min, max } = quantity;
+
+  if (!isAutoscale) {
+    return !exceedsMaxNodes(desired);
+  }
+
+  const isMinValid = min >= 0 && min <= max;
+  const isMaxValid = max <= NODE_RANGE.MAX;
+  const isDesiredInRange = min <= desired && max >= desired;
+
+  return isMinValid && isMaxValid && isDesiredInRange;
+};
+
+export const hasMax5NodesAntiAffinity = (nodePoolState: NodePoolState) =>
+  !nodePoolState.antiAffinity ||
+  (nodePoolState.antiAffinity &&
+    nodePoolState.scaling &&
+    nodePoolState.scaling.quantity.desired <= ANTI_AFFINITY_MAX_NODES);
+
+export const errorWithScalingOrAntiAffinity = (
+  regionInformations: TRegionInformations,
+  nodePoolState: NodePoolState,
+) =>
+  !isScalingValid(nodePoolState) ||
+  !hasMax5NodesAntiAffinity(nodePoolState) ||
+  !zoneAZisChecked(regionInformations, nodePoolState);
+
+export const canToggleAntiAffinity = (
+  state: NodePoolState,
+): state is NodePoolState & {
+  scaling: { isAutoscale: false };
+  antiAffinity: boolean;
+} => state.scaling?.isAutoscale === false;
+
+export const canToggleAutoScale = (
+  state: NodePoolState,
+): state is NodePoolState & {
+  scaling: { isAutoscale: boolean };
+  antiAffinity: false;
+} => state.antiAffinity === false;
```

Comment:
These type guards are misleading: they assert properties without actually checking for them. This can lead to unsafe assumptions in downstream code.

Example: `canToggleAntiAffinity` only checks if `isAutoscale === false` but never if `antiAffinity` is defined in type.

The runtime condition must fully reflect the asserted type to be valid.